File: src/gui/src/profile/models/BankAccountDto.ts
export default class BankAccountDto {
    id: string;
    label: string;
    bankName: string;
    accountHolderName: string;
    iban: string;
    bic: string;
    isDefault: boolean;

    public constructor(
        id: string,
        label: string,
        bankName: string,
        accountHolderName: string,
        iban: string,
        bic: string,
        isDefault: boolean
    ) {
        this.id = id;
        this.label = label;
        this.bankName = bankName;
        this.accountHolderName = accountHolderName;
        this.iban = iban;
        this.bic = bic;
        this.isDefault = isDefault;
    }
}

File: src/backend/WordsmithHub.API/Features/BankAccounts/Models/BankAccountDto.cs
namespace WordsmithHub.API.Features.BankAccounts.Models;

public record BankAccountDto
{
    public required Guid Id { get; set; }
    public required string Label { get; set; }
    public required string BankName { get; set; }
    public required string AccountHolderName { get; set; }
    public required string Iban { get; set; }
    public required string Bic { get; set; }
    public bool IsDefault { get; set; }
}
